pallets/parachain-staking/src/types.rs

```diff
@@ -203,22 +206,27 @@ where
 }
 
 #[derive(Encode, Decode, RuntimeDebug, PartialEq)]
-pub struct Delegator<AccountId: Eq + Ord, Balance: Eq + Ord> {
-	pub delegations: OrderedSet<Stake<AccountId, Balance>>,
+pub struct Delegator<AccountId: Eq + Ord, Balance: Eq + Ord, MaxCollatorsPerDelegator: Get<u32>> {
+	pub delegations: OrderedSet<Stake<AccountId, Balance>, MaxCollatorsPerDelegator>,
 	pub total: Balance,
 }
 
-impl<AccountId, Balance> Delegator<AccountId, Balance>
+impl<AccountId, Balance, MaxCollatorsPerDelegator> Delegator<AccountId, Balance, MaxCollatorsPerDelegator>
 where
 	AccountId: Eq + Ord + Clone + Debug,
 	Balance: Copy + Add<Output = Balance> + Saturating + PartialOrd + Eq + Ord + Debug + Zero,
+	MaxCollatorsPerDelegator: Get<u32> + Debug + PartialEq,
 {
 	pub fn new(collator: AccountId, amount: Balance) -> Self {
 		Delegator {
-			delegations: OrderedSet::from(vec![Stake {
-				owner: collator,
-				amount,
-			}]),
+			delegations: OrderedSet::from(
+				vec![Stake {
+					owner: collator,
+					amount,
+				}]
+				.try_into()
+				.expect("At least one collator per delegator should be enabled"),
```

Comment:
Isn't it a valid use case to set the delegator to 0 and disable delegation by that?

Reply:
The error message war referring to `MaxCollatorsPerDelegator`. But it is correct that we might set this to 0 which would cause a panic. Should be fixed by 1201ef8f7e4aa37cac97308d27b38cd22ac06c43

Reply:
I was maybe thinking that we initialize a Delegator with an empty delegation list and always add a collator explicitly (which could fail). In join_delegators would go through while silently not adding a delegation to the delegator. If i'm not wrong.

there is only one reference to `Delegator::new` so shouldn't be a huge refactor?

Reply:
True. Should be fixed by 522c885